backend/app/api/routes/knowledge_base.py

```
@@ -328,6 +328,55 @@ async def user_info(current_user: str = Depends(get_current_user)):
+            message="Website scraping started in background", 
+            data={
+                "website_url": website_url,
+                "urls_found": len(urls),
+                "unique_urls": len(unique_urls),
+                "urls_to_scrape": len(urls_to_scrape),
+                "batches": (len(urls_to_scrape) + batch_size - 1) // batch_size
+            }
+        )
+    except Exception as e:
+        logger.error(f"Error in scrape_for_setup: {str(e)}", exc_info=True)
+        raise HTTPException(status_code=500, detail=f"Error scraping website: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)